src/layouts/TokenLayout/TokenHolderDetailsCard.tsx

```diff
@@ -0,0 +1,92 @@
+import { useEffect, useState } from 'react';
+import { useSelector } from 'react-redux';
+
+import { ZERO } from 'appConstants';
+import {
+  FormatAmount,
+  FormatUSD,
+  CardItem,
+  AccountLink,
+  CopyButton
+} from 'components';
+import { isValidAccountTokenValue } from 'helpers';
+import { useAdapter, useGetTransactionFilters } from 'hooks';
+import { faUser, faCoins, faDollarSign } from 'icons/solid';
+import { activeNetworkSelector, tokenSelector } from 'redux/selectors';
+import { TokenType } from 'types';
+
+export const TokenHolderDetailsCard = () => {
+  const { id: activeNetworkId } = useSelector(activeNetworkSelector);
+  const { token } = useSelector(tokenSelector);
+  const { identifier } = token;
+  const { sender, receiver } = useGetTransactionFilters();
+  const { getAccountTokens } = useAdapter();
+
+  const [isDataReady, setIsDataReady] = useState<boolean | undefined>();
+  const [accountTokenDetails, setAccountTokenDetails] = useState<TokenType>();
+
+  const filterAddress = sender || receiver;
+  const address = String(filterAddress);
+  const usdValue = accountTokenDetails?.valueUsd ?? ZERO;
+
+  const fetchAccountTokens = async () => {
+    const { data, success } = await getAccountTokens({
```

Comment:
Add error handling for the async fetch.

The fetchAccountTokens function doesn't handle errors that might occur during the API call. If the promise rejects, the error will be unhandled.

Apply this diff to add proper error handling:

 const fetchAccountTokens = async () => {
+  try {
     const { data, success } = await getAccountTokens({
       address,
       identifier,
       includeMetaESDT: true,
       fields: ['balance', 'valueUsd'].join(',')
     });
     if (success && data && data.length <= 1) {
       setAccountTokenDetails(data[0]);
     }
     setIsDataReady(success);
+  } catch (error) {
+    console.error('Failed to fetch account tokens:', error);
+    setIsDataReady(false);
+  }
 };